app/assets/javascripts/discourse/app/lib/range-ref.js

```diff
@@ -0,0 +1,42 @@
+export default class RangeRef {
+  constructor(selector) {
+    this.selector = selector;
+    this.#updateRect();
+    this.setupListeners(selector);
+  }
+
+  #updateRect() {
+    const selection = document.getSelection();
+    this.range = selection && selection.rangeCount && selection.getRangeAt(0);
+    this.rect = this.range.getBoundingClientRect();
+    return this.rect;
+  }
+
+  getBoundingClientRect() {
+    return this.rect;
+  }
+
+  setupListeners(selector) {
+    const update = () => this.#updateRect();
+    document.querySelector(selector).addEventListener("mouseup", update);
+    window.addEventListener("scroll", update);
+    document.scrollingElement.addEventListener("scroll", update);
+  }
+
+  clearListeners() {
```

Comment:
Im on mobile might have missed it. Is it used anywhere?

Reply:
Should work on mobile but only when logged in,
it changes the placement to below so that it doesn't interfere with the default tooltip.


https://github.com/discourse/discourse/blob/150960f174039b65d983258fac220d66dbaa446f/app/assets/javascripts/discourse/app/components/quote-button.js#L191-L199


https://user-images.githubusercontent.com/30090424/228962727-bbe58a17-fc15-43aa-9b41-5931dd2af2fb.MP4

Reply:
No I mean clearListeners, where is it called?

Reply:
Ah yes, I meant to revisit that. I added it here in a previous commit:
https://github.com/discourse/discourse/pull/20878/commits/1069e116a9f224178e596c580556892d592f2e90

But it seems to cause test failures, which I'm not sure why...

Reply:
I think the issue with the initial try is that the the event listeners are registered only after a text selection is started and in cases where a selection hasn't been started, there is no listener, `this.rangeRef` would be `null`, so `this.rangeRef.clearListeners()` failed.